docs/JSONDocumentation.md

```diff
@@ -0,0 +1,379 @@
+| Unsafe        | 1 << 14      | N/A                                           |
+| Partial       | 1 << 15      | N/A                                           |
+
+
+## Field
+
+### Required keys:
+
+| Key  | Type   | .NET              | Java              |
+|------|--------|-------------------|-------------------|
+| Name | string | Name of the field | Name of the field |
+| Type | string | Type of the field | Type of the field |
+
+### Optional keys:
+
+| Key         | Type               | .NET                           | Java                           |
+|-------------|--------------------|--------------------------------|--------------------------------|
+| Initializer | string             | The initial value of the field | The initial value of the field |
+| Modifiers   | integer, minimum 0 | See [Modifiers](#modifiers)    | See [Modifiers](#modifiers)    |
+
+
+## Constructor
+
+### Required keys:
+
+| Key  | Type   | .NET                    | Java                    |
+|------|--------|-------------------------|-------------------------|
+| Name | string | Name of the constructor | Name of the constructor |
+
+### Optional keys:
+
+| Key                    | Type                                              | .NET                                       | Java                         |
+|------------------------|---------------------------------------------------|--------------------------------------------|------------------------------|
+| Parameters             | array of [Parameter](#parameter)                  |                                            |                              |
+| Statements             | array of [Statement](#statement)                  |                                            |                              |
+| Modifiers              | integer, minimum 0                                | See [Modifiers](#modifiers)                | See [Modifiers](#modifiers)  |
+| Attributes             | array of [Attribute](#attribute)                  |                                            | Annotations                  |
+| DocumentationComments  | [DocumentationComments](#documentation-comments)  | XML documentation comments                 | JavaDoc                      |
+
+
+## Method
+
+### Required keys:
+
+| Key  | Type   | .NET               | Java               |
+|------|--------|--------------------|--------------------|
+| Name | string | Name of the method | Name of the method |
+
+### Optional keys:
+
+| Key                   | Type                                             | .NET                                       | Java                         |
+|-----------------------|--------------------------------------------------|--------------------------------------------|------------------------------|
+| ReturnType            | string                                           |                                            |                              |
```

Comment:
Is a method that doesn't return anything supposed to have `ReturnType` missing, or `ReturnType` present but with the string `"void"`?

Reply:
Missing. I'll specify that!

Reply:
Do we also need an issue to change our analyzer to match? Or does it already do the right thing?

Reply:
Looking at our analyzer's output for the different examples, those do seem to contain `"ReturnType": "void",` so we'll need a (probably small) fix to make this right.

Submitted as #110.